### PR TITLE
Java16

### PR DESCRIPTION
**Pull request by arseniimalov**

### CODE REVIEW

File: smartclient-core/src/main/java/org/srg/smartclient/JDBCHandlerFactory.java [hunk 1]
```diff
@@ -146,6 +146,11 @@ protected static void applySmartClientFieldAnnotation(SmartClientField sfa, DSFi
             if (!sfa.customSelectExpression().isBlank()) {
                 dsf.setCustomSelectExpression(sfa.customSelectExpression());
             }
+            if (sfa.canEdit()) {
```
Comment (kdavydiuk) [inline smartclient-core/src/main/java/org/srg/smartclient/JDBCHandlerFactory.java]: why not to replace it with `dsf.setCanEdit(sfa.canEdit())`?

File: smartclient-core/src/main/java/org/srg/smartclient/utils/AnnotationUtils.java [hunk 2]
```diff
@@ -105,14 +106,88 @@ private static String replaceLast(String text, String regex, String replacement)
...
+
+    /**
+     * Retrieves all annotation values.
+     */
+    public  static <A extends Annotation> Map<String, Object> annotationValues(A annotation) {
+        final Map<String, Object> values = new HashMap<>();
+
+        final  Method methods[] =  annotation.annotationType().getDeclaredMethods();
+        for (Method m : methods) {
+            Object v = null;
```
Comment (kdavydiuk) [inline smartclient-core/src/main/java/org/srg/smartclient/utils/AnnotationUtils.java]: Seems like you don't use object `v` outside of try block. I think it's better to create and init it in line 173.

File: pom.xml [hunk 3]
```diff
@@ -4,12 +4,12 @@
 
 	<groupId>org.srg.smartclient-dsi</groupId>
 	<artifactId>smartclient-parent</artifactId>
-	<version>1.0</version>
+	<version>1.0-SNAPSHOT</version>
```
Comment (kdavydiuk) [inline pom.xml]: Why did you change the version back to SNAPSHOT?

**Review by kdavydiuk — APPROVED** [review]

**Review by kovantonlenko — APPROVED** [review]
